### DISCUSSION TITLE
Problems with code learning

### DISCUSSION THREAD

**Discussion by MayueCif** [elsa-workflows/elsa-core#4421] [Q&A]
I'm reading the elsa code, but the project is too complex, I want to see the part of the code where the workflow is executed and published after Docker starts the project, but I'm only reading up to the mediator send event here, and I can't find the corresponding handler. is there no code for this part? 
```
await _mediator.Publish(new WorkflowDefinitionPublishing(workflowDefinition), cancellationToken);
await _workflowDefinitionStore.SaveAsync(workflowDefinition, cancellationToken);
await _mediator.Publish(new WorkflowDefinitionPublished(workflowDefinition), cancellationToken);
```

**Comment by sfmskywalker** [comment]
Hello @MayueCif,

Thank you for your question!

1. When you publish a workflow definition, it's not executed but simply saved in the persistence store.
2. I understand that following the code path can be a bit tricky with eventing. But with some practice, it'll become clearer. To find the handler(s) for notifications like `WorkflowDefinitionPublishing` and `WorkflowDefinitionPublished`, you can use your IDE's "find usages" tool. I personally use JetBrains Rider, and here's how it looks:
   
   ![Find Usages Tool](https://github.com/elsa-workflows/elsa-core/assets/938393/00c00565-1d6f-473c-9171-217fdd807929)

   As shown, there are 3 usages. To find the handler of the notification, navigate to its `Handle` method.
3. Remember, Elsa is open-ended. Some notifications might not be handled within Elsa but are there for external codes, like your application, to handle specific events.

Hope this helps! Let me know if you have any more questions.
